src/components/widgets/Map.vue

```diff
@@ -316,6 +327,75 @@ onMounted(async () => {
     }
   })
 
+  const saveCtl = savetiles(esri, {
+    saveWhatYouSee: true,
+    maxZoom: 19,
+    alwaysDownload: false,
+    position: 'topright',
+    parallel: 20,
+    confirm(status: SaveStatus, ok: () => void) {
+      showDialog({
+        variant: 'info',
+        message: `Save ${status._tilesforSave.length} tiles for offline use?`,
+        persistent: false,
+        timer: undefined,
+        maxWidth: '450px',
+        actions: [
+          { text: 'Cancel', color: 'white', action: closeDialog },
+          {
+            text: 'Save tiles',
+            color: 'white',
+            action: () => {
+              ok()
+              closeDialog()
+            },
+          },
+        ] as DialogActions[],
+      })
+    },
+    confirmRemoval(_status: SaveStatus, ok: () => void) {
+      showDialog({
+        variant: 'warning',
+        message: 'Remove all saved tiles for this layer?',
+        persistent: false,
+        timer: undefined,
+        maxWidth: '450px',
+        actions: [
+          { text: 'Cancel', color: 'white', action: closeDialog },
+          {
+            text: 'Remove tiles',
+            color: 'white',
+            action: () => {
+              ok()
+              closeDialog()
+              openSnackbar({ message: 'Offline tiles removed', variant: 'info', duration: 3000 })
+            },
+          },
+        ] as DialogActions[],
+      })
+    },
+    saveText: '<i class="mdi mdi-download" title="Save tiles"></i>',
+    rmText: '<i class="mdi mdi-trash-can" title="Remove tiles"></i>',
+  })
+
+  if (map.value) {
+    saveCtl.addTo(map.value)
+  }
+
+  esri.on('savestart', (e: any) => {
```

Comment:
From this part of the code I assume the saving events are attached specific to the ESRI tiles, correct? That means, for example, that if the ESRI tiles downloading finish, but the others do not, and the user closes cockpit assuming they are all downloaded, the others will be not, right?

Can we do it in a way that it waits for all of them to be downloaded?